packages/monitoring-experience-api/src/middleware/ipWhitelistValidation.ts

```diff
@@ -0,0 +1,43 @@
+import type { AbstractController, UnknownRequest } from '@verkkokauppa/core'
+import ipRangeCheck from 'ip-range-check'
+import { IpAddressValidationError } from '../errors'
+
+export const withIpWhitelistValidation = <
+  TController extends new (...args: any[]) => AbstractController
+>(
+  Controller: TController
+): TController => {
+  abstract class IpAddressWhitelistValidationController extends Controller {
+    protected async validateRequest(req: UnknownRequest) {
+      const validIps = process.env.ROUTE_PUBLIC_MONITORING_IP_WHITELIST
+        ? process.env.ROUTE_PUBLIC_MONITORING_IP_WHITELIST.split(' ')
+        : ['127.0.0.1']
+
+      const requestClientIp =
+        req.headers['x-forwarded-for'] || req.socket?.remoteAddress
+
+      if (!requestClientIp) {
+        throw new IpAddressValidationError('No IP found to validate')
+      }
+
+      let parsedClientIp
```

Comment:
Could you add some comments what is happening in these splits, its hard to see why these are happenin.

Could you also write some kind of automation test and move this parsin functionality to own service?
Like UrlParseUtil etc